lib/jxl/dec_frame.cc

```diff
@@ -816,6 +831,34 @@ Status FrameDecoder::ProcessSections(const SectionInfo* sections, size_t num,
     }
     FinalizeDC();
     AllocateOutput();
+    if (pause_at_progressive_ && !single_section) {
+      bool can_return_dc = true;
+      if (single_section) {
+        // If there's only one group and one pass, there is no separate section
+        // for DC and the entire full resolution image is available at once.
+        can_return_dc = false;
+      }
+      if (!decoded_->metadata()->extra_channel_info.empty()) {
+        // If extra channels are encoded with modular without squeeze, they
+        // don't support DC. If the are encoded with squeeze, DC works in theory
+        // but the implementation may not yet correctly support this for Flush.
+        // Therefore, can't correctly pause for a progressive step if there is
+        // an extra channel (including alpha channel)
+        can_return_dc = false;
```

Comment:
It should be possible at this point to know whether the modular channel(s) can be shown or not, right?

Can keep it like this for this PR, but would be good to have the actual condition in here. Alpha does use squeeze by default in VarDCT mode so it's not like it's a rare case.

Reply:
Does the current implementation of FrameDecoder::Flush handle squeeze correctly? At least alpha didn't work in my testing (with relatively default compression settings), but something else may have been wrong

Reply:
When I try it with `decode_progressive` (https://github.com/libjxl/libjxl/pull/606), things do seem to work, yes. When did you do your testing? Things were broken before https://github.com/libjxl/libjxl/pull/629 was merged.